python/tests/test_shared_pipeline0.py

```diff
@@ -0,0 +1,604 @@
+        CREATE VIEW v_decimal AS SELECT c1 + 2.75::DECIMAL(5, 2) as c1 FROM tbl_decimal;
+        """
+        from decimal import Decimal
+
+        data = [{"c1": 2.25}]
+        expected = [{"c1": Decimal("5.00"), "insert_delete": 1}]
+        self.pipeline.start()
+        out = self.pipeline.listen("v_decimal")
+        self.pipeline.input_json("tbl_decimal", data=data)
+        self.pipeline.wait_for_completion(True)
+        got = out.to_dict()
+        assert expected == got
+
+    def test_pandas_array(self):
+        """
+        CREATE TABLE tbl_array (c1 INT ARRAY);
+        CREATE VIEW v_array AS SELECT c1 FROM tbl_array;
+        """
+        data = [{"c1": [1, 2, 3]}]
+        out = self.pipeline.listen("v_array")
+        self.pipeline.start()
+        self.pipeline.input_json("tbl_array", data=data)
+        self.pipeline.wait_for_completion(True)
+        got = out.to_dict()
+        expected = [{"c1": [1, 2, 3], "insert_delete": 1}]
+        assert got == expected
+
+    def test_pandas_struct(self):
+        """
+        CREATE TYPE s_struct AS (
+            f1 INT,
+            f2 STRING
+        );
+        CREATE TABLE tbl_struct (c1 s_struct);
+        CREATE VIEW v_struct AS SELECT c1 FROM tbl_struct;
+        """
+        data = [{"c1": {"f1": 1, "f2": "a"}}]
+        self.pipeline.start()
+        out = self.pipeline.listen("v_struct")
+        self.pipeline.input_json("tbl_struct", data)
+        self.pipeline.wait_for_completion(True)
+        got = out.to_dict()
+        expected = [{"c1": {"f1": 1, "f2": "a"}, "insert_delete": 1}]
+        assert got == expected
+
+    def test_pandas_date_time_timestamp(self):
+        """
+        CREATE TABLE tbl_datetime (c1 DATE, c2 TIME, c3 TIMESTAMP);
+        CREATE VIEW v_datetime AS SELECT c1, c2, c3 FROM tbl_datetime;
+        """
+        from pandas import Timestamp, Timedelta
+
+        data = [{"c1": "2022-01-01", "c2": "12:00:00", "c3": "2022-01-01 12:00:00"}]
+        expected = [
+            {
+                "c1": Timestamp("2022-01-01 00:00:00"),
+                "c2": Timedelta("0 days 12:00:00"),
+                "c3": Timestamp("2022-01-01 12:00:00"),
+                "insert_delete": 1,
+            }
+        ]
+        self.pipeline.start()
+        out = self.pipeline.listen("v_datetime")
+        self.pipeline.input_json("tbl_datetime", data)
+        self.pipeline.wait_for_completion(True)
+        got = out.to_dict()
+        assert expected == got
+
+    def test_pandas_simple(self):
+        """
+        CREATE TABLE tbl_simple (
+            c0 BOOLEAN, c1 TINYINT, c2 SMALLINT, c3 INT, c4 BIGINT, c5 REAL, c6 DOUBLE, c7 VARCHAR, c8 CHAR
+        );
+        CREATE VIEW v_simple AS SELECT * FROM tbl_simple;
+        """
+        data = [
+            {
+                "c0": True,
+                "c1": 1,
+                "c2": 2,
+                "c3": 3,
+                "c4": 4,
+                "c5": 5.0,
```

Comment:
so is the decimal bug fixed?

Reply:
No, the decimal bug is for adhoc queries. Here when listening to a view, we have its schema, so FPs are treated correctly.